pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -481,6 +481,13 @@ func (m *kubeGenericRuntimeManager) computePodActions(pod *v1.Pod, podStatus *ku
 				message := fmt.Sprintf("Container %+v is dead, but RestartPolicy says that we should restart it.", container)
 				glog.Info(message)
 				changes.ContainersToStart = append(changes.ContainersToStart, idx)
+			} else {
```

Comment:
I think we can call PostStopContainer unconditionally here, since at this point the container ID doesn't refer to a running container. This could prevent the next container restart from failing due to inability to allocate CPUs (because of a race between this sync and the garbage collector.)